java-jacoco/src/main/java/org/sonar/plugins/jacoco/AbstractAnalyzer.java

```diff
@@ -143,22 +134,20 @@ private void populateClassFilesCache(File dir, String path) {
   }
 
   private void readExecutionData(File jacocoExecutionData, SensorContext context) {
-    ExecutionDataVisitor executionDataVisitor = new ExecutionDataVisitor();
-    boolean useCurrentFormat = false;
     if (jacocoExecutionData == null || !jacocoExecutionData.isFile()) {
       JaCoCoExtensions.LOG.info("Project coverage is set to 0% as no JaCoCo execution data has been dumped: {}", jacocoExecutionData);
       jacocoExecutionData = null;
-    } else {
-      try {
-        useCurrentFormat = readJacocoReport(jacocoExecutionData, executionDataVisitor);
-      } catch (IOException e) {
-        throw new SonarException(e);
-      }
+    }
+    ExecutionDataVisitor executionDataVisitor = new ExecutionDataVisitor();
+    try {
+      jacocoReportReader = new JacocoReportReader(jacocoExecutionData).readJacocoReport(executionDataVisitor, executionDataVisitor);
+    } catch (IOException e) {
+      throw new SonarException("An error occured while reading JaCoCo report", e);
```

Comment:
@benzonico once again: error message about failure to read file is inconsistent - see JaCoCoReportMerger. Code can be simplified, if you'll pull catch of IOException down into JacocoReportReader - less places, where exception should be handled; single place, where message will be constructed.